packages/flutter_tools/lib/src/ios/xcresult.dart

```diff
@@ -236,3 +260,47 @@ enum XCResultIssueType {
   /// This is for all the issues under the `errorSummaries` key in the xcresult.
   error,
 }
+
+// A typical location url string looks like file:///foo.swift#CharacterRangeLen=0&EndingColumnNumber=82&EndingLineNumber=7&StartingColumnNumber=82&StartingLineNumber=7.
+//
+// This function converts it to something like: file:///foo.swift:<StartingLineNumber>:<StartingColumnNumber>.
+String _convertUrlToLocationString(String url){
+  if (url.isEmpty) {
+    return '';
+  }
+  final List<String> topComponents = url.split('#');
```

Comment:
We shouldn't parse `Uri`s by hand.  Unfortunately Xcode decided to put the query key values into the fragment, but we can swap them:
```dart
// A typical location url string looks like file:///foo.swift#CharacterRangeLen=0&EndingColumnNumber=82&EndingLineNumber=7&StartingColumnNumber=82&StartingLineNumber=7.
//
// This function converts it to something like: foo.swift:<StartingLineNumber>:<StartingColumnNumber>.
String? _convertUrlToLocationString(String url) {
  final Uri? fragmentLocation = Uri.tryParse(url);
// Maybe log something here
  if (fragmentLocation == null) {
    return null;
  }
  // Parse the fragment as a query of key-values:
  final Uri fileLocation = Uri(
    path: fragmentLocation.path,
    query: fragmentLocation.fragment,
  );

  String? startingLineNumber = fileLocation.queryParameters['StartingLineNumber'] ?? '';
  if (startingLineNumber.isNotEmpty) {
    startingLineNumber = ':$startingLineNumber';
  }
  String startingColumnNumber = fileLocation.queryParameters['StartingColumnNumber'] ?? '';
  if (startingColumnNumber.isNotEmpty) {
    startingColumnNumber = ':$startingColumnNumber';
  }

  return '${fileLocation.path}$startingLineNumber$startingColumnNumber';
}
```